### PR TITLE
Implement link-type update API(B-01~03)

### PR DESCRIPTION
**Pull request by kachick**

### CODE REVIEW

File: lib/sbpayment/link/update_report.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
+require_relative '../parameter_definition'
+
+module Sbpayment
+  module Link
+    # B03-1
+    class UpdateReport
```
Comment (kachick) [inline lib/sbpayment/link/update_report.rb]: I don't have opinions for `Update`

File: spec/sbpayment/link/update_result_spec.rb [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+    @result = Sbpayment::Link::UpdateResult.new
+    @result.update_attributes params, utf8: true
+  end
+end
+
+describe Sbpayment::Link::UpdateResult do
+  context 'when valid params are given' do
+    include_context "prepare update result params" do
+      let!(:query) { 'res_payinfo_key=&free1=&cust_code=498b1fd98ebeb074ff1d50f2e2720a1a&free3=&res_sps_payment_no=&free2=&sps_hashcode=6b7981691587c72f94def4c7192e7b8c6d624b85&res_date=20150807161021&res_err_code=&limit_second=600&pay_type=2&res_payment_date=20150807161021&merchant_id=19788&request_date=20150807161004&res_pay_method=&terminal_type=0&res_sps_cust_no=&pay_method=&sps_payment_no=&res_result=OK&sps_cust_no=' }
```
Comment (kachick) [inline spec/sbpayment/link/update_result_spec.rb]: Correct? Please check!

**Comment by kachick** [comment]
Many code imported from A01~A03. When the spec is correct, I'll add `update_report_spec` as these. @banyan Please review!

**Comment by banyan** [comment]
:+1:

**Pull request merged**